pkg/expr/sql/db.go

```diff
@@ -152,21 +152,20 @@ func convertDataType(fieldType data.FieldType) mysql.Type {
 	case data.FieldTypeTime:
 		return types.Timestamp
 	default:
-		fmt.Println("------- Unsupported field type: ", fieldType)
+		fmt.Printf("------- Unsupported field type: %t", fieldType)
 		return types.JSON
 	}
 }
 
 func (db *DB) QueryFramesInto(tableName string, query string, frames []*data.Frame, f *data.Frame) error {
-
 	pro := memory.NewDBProvider(db.inMemoryDb)
 	session := memory.NewSession(mysql.NewBaseSession(), pro)
 	ctx := mysql.NewContext(context.Background(), mysql.WithSession(session))
 
 	for _, frame := range frames {
 		// We have both `frame` and `f` in this function. Consider renaming one or both.
 		// Potentially `f` to `outputFrame`
-		err := db.writeDataframeToDb(ctx, frame.Name, frame)
+		err := db.writeDataframeToDb(ctx, tableName, frame)
```

Comment:
I'm pretty sure this is a bad change. I think you want `frame.RefID` here.

If you think about it, we're inside a loop and looping over all the input dataframes. We want to write each one to the database. So each one will need to be giving a different table name (we can't write two frames to the same table with the same name). So we can't use `tableName` here, as it doesn't change between frames.

I think adding tests as you go can really help **me** to understand what functionality is now passing, and stop me from regressing some functionality that you feel you have added. So I'd really like us to consider continuing to add tests as we go. If we don't do this, I expect we'll be playing whack-a-mole, and each of us will be fixing different things, but sometimes breaking one-anothers' fixes.

Reply:
The "tableName" variable here should be RefID I think. 

My thinking is that it is going to be one Frame Per RefID. Although that is not that query response we get in (there can be multiple frames) - I imagined in that case we will do a conversion (e.g. Multi -> Long) into a single Frame to make it more understandable by SQL

(In the cases where there are multiple frames, per refId, it is usually more one would reference directly. There is also nothing other than their order to guarantee any sort of uniqueness. So an alternative would need work with that, or reject data with different table names.

Reply:
ok thanks. I'll have another look at this. After writing my first message I decided I must be wrong anyway - the screenshot you shared yesterday showed that you were able to JOIN two tables - so of course they must have been written to the database with different table-names to make that possible!